OPAL/br/src/main/scala/org/opalj/br/fpcf/properties/Alias.scala

```diff
@@ -22,15 +22,53 @@ sealed trait AliasPropertyMetaInformation extends PropertyMetaInformation {
  * An alias property provides information about the relationship of the memory locations of the associated entities.
  *
  *  - [[NoAlias]] can be assigned to a pair of entities, iff no execution path exists where both entities refer to the same memory location.
+ *   This implies that there is a definite separation of the memory locations of the two entities, and no interference can occur when one of the entities is referenced.
+ *   Note that this does not imply that, for example, the fields of the entities cannot refer to the same memory location.
+ *   Example:
+ *   {{{
+ *     Object m(Object a) {
+ *       Object b = new Object();
+ *       return b;
+ *     }
+ *   }}}
+ *   In this example, the formal parameter `a` and the local variable `b` are guaranteed to never refer to the same memory location,
+ *   as `b` is always assigned to a new object that the formal parameter can't possibly be pointing to since the object has not been created before the method call.
+ *   This ensures that the method's return value and the formal parameter never point to the same memory location.
  *
  *  - [[MustAlias]] can be assigned to a pair of entities, iff both entities refer to the same memory location in every execution time at all times.
- * This implies that at every use site of one entity, it can be replaced with the other entity (if it is defined at the current location)
+ *   This implies that at every use site of one entity, it can be replaced with the other entity (if it is defined at the current location) without changing the semantics of the program.
+ *   Example:
+ *   {{{
+ *    Object m() {
+ *      Object a = new Object();
+ *      Object b = a;
+ *      return b;
+ *    }
+ *   }}}
+ *   In this example, the local variable `a` and `b` are guaranteed to refer to the same memory location at all times,
+ *   because a is only assigned once and b is only defined once with the value of a.
+ *   This means that the return value of the method can be replaced with `a`.
```

Comment:
True, but wouldn't the above definition imply that one could also replace a with b? Now, there is only one use site of a, in line 44, and if there were more they would have to be after line 44 and thus replacable (use sites before line 44 should make the two variables clearly MayAlias), but replacing a with b in line 44 obviously can't work.

Reply:
Yes, but the definition (regarding replacement) above states that it is only applicable if the other entity (b in this case) is defined at that location, which is not given for the right-hand side of line 44.